build.py

```
@@ -0,0 +1,435 @@
+    dist_dir = Path("dist")
+    if not dist_dir.exists():
+        print("❌ Dist-Verzeichnis nicht gefunden")
+        return
+    
+    # UPX-Komprimierung (falls verfügbar)
+    try:
+        exe_files = list(dist_dir.glob("*.exe"))
+        for exe_file in exe_files:
+            original_size = exe_file.stat().st_size
+            
+            subprocess.run(["upx", "--best", str(exe_file)], 
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)